File: src/main/java/com/teamwizardry/wizardry/common/spell/component/ShapeInstance.java
package com.teamwizardry.wizardry.common.spell.component;

import java.util.Map;

public class ShapeInstance extends Instance
{
    public ShapeInstance(Pattern pattern, TargetType targetType, Map<String, Double> attributeValues, double manaCost, double burnoutCost, Interactor caster)
    {
        super(pattern, targetType, attributeValues, manaCost, burnoutCost, caster);
        extraData.putString(PATTERN_TYPE, "shape");
    }
}

File: src/main/kotlin/com/teamwizardry/wizardry/common/spell/component/ShapeInstance.kt
package com.teamwizardry.wizardry.common.spell.component

class ShapeInstance(
    pattern: Pattern,
    targetType: TargetType,
    attributeValues: Map<String, Double>,
    manaCost: Double,
    caster: Interactor
) : Instance(pattern, targetType, attributeValues, manaCost, caster) {
    init {
        extraData.putString("pattern_type", "shape")
    }
}